src/audio/mixer.c

```diff
@@ -174,25 +174,21 @@ static int mixer_params(struct comp_dev *dev,
 {
 	struct sof_ipc_comp_config *config = COMP_GET_CONFIG(dev);
 	struct comp_buffer *sinkb;
-	struct comp_buffer *sourceb;
 	uint32_t period_bytes;
 
 	trace_mixer_with_ids(dev, "mixer_params()");
 
-	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
```

Comment:
@bkokoszx 
I checked with the SRC pipeline with mixer
If I start SRC PCM, the SRC pipeline with mixer can work.
If I start SRC PCM first then the normal PCM, both of them can work.
As mixer will have multiple source, and now the buffer size is set up during HW param, maybe you should try to have some logic to find the active souce instead of use the single sink instead.

PS: You may also need to check the mux and de-mux comp.

Reply:
@xiulipan
I'm not sure whether we need ATM to add logic and search for "active" source since sources and sinks would have same frame_fmt (in my opinion for mixer sink buffer is sufficient). This could be refined during #1280.

I will take a look at mux and demux.